dr_ensenso_node/src/ensenso.cpp

```diff
@@ -483,6 +487,64 @@ class EnsensoNode: public ros::NodeHandle {
 		return true;
 	}
 
+	bool onGetCameraParams(dr_ensenso_msgs::GetCameraParams::Request & req, dr_ensenso_msgs::GetCameraParams::Response & res) {
+		if (true
+			&& (req.camera) == dr_ensenso_msgs::GetCameraParams::Request::MONO
+			&& (req.camera) == dr_ensenso_msgs::GetCameraParams::Request::LEFT
+			&& (req.camera) == dr_ensenso_msgs::GetCameraParams::Request::RIGHT
+		) {
+			ROS_ERROR_STREAM("Invalid camera type " << req.camera << " requested.");
+			return false;
+		}
+
+		if (req.camera == dr_ensenso_msgs::GetCameraParams::Request::MONO && !ensenso_camera->hasMonocular()) {
+			ROS_ERROR_STREAM("No monocular camera available!");
+			return false;
+		}
+
+		cv::Mat mat;
+		switch (req.type) {
+			case dr_ensenso_msgs::GetCameraParams::Request::CAMERA_MATRIX: {
+				NxLibItem item = (req.camera == dr_ensenso_msgs::GetCameraParams::Request::MONO)
+				? (*(ensenso_camera->nativeMonocular()))[itmCalibration]
+				: ensenso_camera->native()[itmCalibration][itmMonocular][req.camera == "Left" ? itmLeft : itmRight];
```

Comment:
I'm not really a fan of this type of usage of the ternary operator. If this switch is moved to its own function (see above), then you can also rewrite the ternary operator into a few `if (...) return ...;` statement. I think that will be much more readable.